### PR TITLE
Prometheus: Move existing query logic to new buffered package

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**:

This is the first step in a series of PRs that will use the prometheus JSON streaming parser added by https://github.com/grafana/grafana/pull/44520.

The current plan is:
1. Move existing logic to `buffered` package (this PR) 
2. Add separate `querydata` package that uses a standard HTTP client instead of the prometheus go client and the streaming JSON response parser (https://github.com/grafana/grafana/pull/48477)
3. Remove `buffered` package & feature flag (future PR once things have stabilized)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/tsdb/prometheus/buffered/types.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import (
+	"time"
+
+	apiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+type clientGetter func(map[string]string) (apiv1.API, error)
+
+type PrometheusQuery struct {
```
Comment (ryantxu) [inline pkg/tsdb/prometheus/buffered/types.go:11]: how does this relate to:
pkg/tsdb/prometheus/types.go

Comment (toddtreece) [reply pkg/tsdb/prometheus/buffered/types.go:11]: good catch. i missed removing `pkg/tsdb/prometheus/types.go`

**Comment by ryantxu** [comment]
Are there any behavioral changes here?  It looks like tracer is passed around differently, but I can not tell if it adds up to the same thing :)

Given the changes required for the raw json parsing, do we *need* a different version of testdata?  and types.go?

**Comment by toddtreece** [comment]
> Are there any behavioral changes here? It looks like tracer is passed around differently, but I can not tell if it adds up to the same thing :)

@ryantxu there shouldn't be any behavioral changes with this one. all of the minor changes are related to passing around dependencies a little differently due to the move to a separate package

> Given the changes required for the raw json parsing, do we need a different version of testdata? and types.go?

good call on testdata, i moved that back and updated the path in the test. i think there are enough changes to `types.go` in the next PR to keep it as a separate file in `buffered`

**Comment by ryantxu** [comment]
I like the approach -- I think we *could* find a way to do this without exposing public functions, but also don't think it matters enough to worry about as part of a migration path.

For this PR, the @grafana/observability-metrics needs to feel comfortable that this does not introduce any real changes, just structures things for easier comparison in the *very near* future.

**Review by gabor — APPROVED** [review]
LGTM!

**Pull request merged**